src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -71,6 +71,15 @@ public function load(array $configs, ContainerBuilder $container)
         $container->setParameter('kernel.trusted_proxies', $config['trusted_proxies']);
         $container->setParameter('kernel.default_locale', $config['default_locale']);
 
+        if (isset($config['ip_retriever']) && null !== $config['ip_retriever']) {
+            if ($container->has($config['ip_retriever']) === false) {
```

Comment:
this is broken: only services defined in Frameworkbundle are available at that point